WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/CategoryViewHolder.kt

```diff
@@ -1,31 +1,28 @@
 package org.wordpress.android.ui.layoutpicker
 
-import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
-import kotlinx.android.synthetic.main.modal_layout_picker_category.view.*
-import org.wordpress.android.R
+import org.wordpress.android.databinding.ModalLayoutPickerCategoryBinding
 import org.wordpress.android.login.util.getColorStateListFromAttribute
 import org.wordpress.android.util.getColorFromAttribute
 import org.wordpress.android.util.setVisible
+import org.wordpress.android.util.viewBinding
 
 /**
  * Renders the Layout Category header buttons
  */
-class CategoryViewHolder(internal val parent: ViewGroup) : RecyclerView.ViewHolder(
-        LayoutInflater.from(parent.context).inflate(
-                R.layout.modal_layout_picker_category,
-                parent,
-                false
-        )
-) {
-    val container: View = itemView.category_container
-    val category: TextView = itemView.category
-    val emoji: TextView = itemView.emoji
-    val check: ImageView = itemView.check
+class CategoryViewHolder(
+    internal val parent: ViewGroup,
+    internal val binding: ModalLayoutPickerCategoryBinding =
```

Comment:
Suggestion (💡): I like what you did here, but I would consider using another pattern, the one that is already being utilised for `ViewHolder` classes in our codebase (for example, see `TimezoneHeaderViewHolder` and `TimezoneViewHolder`).

Based on this pattern you need to create a `companion object` with a singe `from(...)` function like the below:
```
    companion object {
        fun from(parent: ViewGroup): CategoryViewHolder {
            val binding = ModalLayoutPickerCategoryBinding.inflate(
                    LayoutInflater.from(parent.context),
                    parent,
                    false
            )
            return CategoryViewHolder(parent, binding)
        }
    }
```

Then, `CategoryViewHolder`'s constructor will end up being:
```
class CategoryViewHolder(
    private val parent: ViewGroup,
    private val binding: ModalLayoutPickerCategoryBinding
) : RecyclerView.ViewHolder(binding.root) {
```

And, the only change required from the usage point of view is for the `CategoriesAdapter` to switch from `CategoryViewHolder(parent)` to `CategoryViewHolder.from(parent)`.

Wdyt about keep using this pattern instead?

PS: The exact same applies for the `LayoutViewHolder` class.

Reply:
@ParaskP7 - Interesting, I didn't look at `TimezoneHeaderViewHolder` or `TimezoneViewHolder`, just wrote as is. Either or works for me. Thanks

Reply:
👋 @zwarm !

Great, yes, and as we were discussing about all that these day, I am only in favor of doing this change to keep consistency and help others reuse this pattern when the same need arise again, that is instead of choosing one over the other and losing time in figuring it out in between.

Reply:
@ParaskP7 - That's fine, I don't disagree. I already made the change in commit c654b23. Just saying that neither pattern is wrong.

Reply:
Updated in 2e428c2